### PR TITLE
Implement preact/inferno SSR

### PR DESCRIPTION
**Pull request by timneutkens**
Last night @rauchg had the idea to use the module-alias module used here: https://github.com/zeit/next.js/pull/767 and implement it with the programmatic api. This PR does that :smile:.

### CODE REVIEW

**Comment by timneutkens** [comment]
```
preact:
Concurrency Level:      10
Time taken for tests:   56.893 seconds
Complete requests:      1000
Failed requests:        0
Total transferred:      290613000 bytes
HTML transferred:       290452000 bytes
Requests per second:    17.58 [#/sec] (mean)
Time per request:       568.930 [ms] (mean)
Time per request:       56.893 [ms] (mean, across all concurrent requests)
Transfer rate:          4988.34 [Kbytes/sec] received

Inferno
Concurrency Level:      10
Time taken for tests:   87.391 seconds
Complete requests:      1000
Failed requests:        0
Total transferred:      329434000 bytes
HTML transferred:       329273000 bytes
Requests per second:    11.44 [#/sec] (mean)
Time per request:       873.909 [ms] (mean)
Time per request:       87.391 [ms] (mean, across all concurrent requests)
Transfer rate:          3681.31 [Kbytes/sec] received

React:
Concurrency Level:      10
Time taken for tests:   242.178 seconds
Complete requests:      1000
Failed requests:        0
Total transferred:      1378463000 bytes
HTML transferred:       1378301000 bytes
Requests per second:    4.13 [#/sec] (mean)
Time per request:       2421.783 [ms] (mean)
Time per request:       242.178 [ms] (mean, across all concurrent requests)
Transfer rate:          5558.53 [Kbytes/sec] received
```

File: examples/using-preact/next.config.js [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
     }
+
+    // Disable uglify. This has been fixed in https://github.com/developit/preact-compat/issues/155.
+    // Can be removed once there is a new preact-compat release.
+    config.plugins = config.plugins.filter((plugin) => {
+      if (plugin.constructor.name === 'UglifyJsPlugin') {
+        return false
+      } else {
+        return true
+      }
```
Comment (frol) [inline examples/using-preact/next.config.js]: How about simplifying this to:

```javascript
config.plugins = config.plugins.filter((plugin) => (plugin.constructor.name !== 'UglifyJsPlugin'))
```

or at least drop the redundant if/return true/false:

```javascript
config.plugins = config.plugins.filter((plugin) => {
    return plugin.constructor.name !== 'UglifyJsPlugin'
})
```

Comment (timneutkens) [inline examples/using-preact/next.config.js]: Good catch. I copied this one: https://github.com/zeit/next.js/issues/1195#issuecomment-280986377

Comment (frol) [inline examples/using-preact/next.config.js]: Hmm... "Good catch. I don't care." Was it the case?

Comment (timneutkens) [inline examples/using-preact/next.config.js]: Nope, didn't get to changing it before rauchg merged it 😅 Will send a PR 👍🏻👍🏻

Comment (timneutkens) [inline examples/using-preact/next.config.js]: https://github.com/zeit/next.js/pull/1351

**Pull request merged**